perfmetrics/scripts/run_e2e_tests.sh

```diff
@@ -53,28 +53,162 @@ sudo apt-get install -y gcc python3-dev python3-setuptools
 sudo apt install -y python3-crcmod
 
 # Create bucket for integration tests.
-# The prefix for the random string
-bucketPrefix="gcsfuse-integration-test-"
-# The length of the random string
-length=5
-# Generate the random string
-random_string=$(tr -dc 'a-z0-9' < /dev/urandom | head -c $length)
-BUCKET_NAME=$bucketPrefix$random_string
-echo 'bucket name = '$BUCKET_NAME
-# We are using gcloud alpha because gcloud storage is giving issues running on Kokoro
-gcloud alpha storage buckets create gs://$BUCKET_NAME --project=$PROJECT_ID --location=$BUCKET_LOCATION --uniform-bucket-level-access
+function create_bucket() {
+  bucketPrefix=$1
+  # The length of the random string
+  length=5
+  # Generate the random string
+  random_string=$(tr -dc 'a-z0-9' < /dev/urandom | head -c $length)
+  BUCKET_NAME=$bucketPrefix$random_string
+  # We are using gcloud alpha because gcloud storage is giving issues running on Kokoro
+  gcloud alpha storage buckets create gs://$BUCKET_NAME --project=$PROJECT_ID --location=$BUCKET_LOCATION --uniform-bucket-level-access
+  echo $BUCKET_NAME
+}
+
+# Non parallel execution of integration tests located within specified test directories for group 1.
+function run_non_parallel_tests_group_1() {
+  for test_dir_np_group_1 in "${test_dir_non_parallel_group_1[@]}"
+  do
+    test_path_non_parallel_group_1="./tools/integration_tests/$test_dir_np_group_1"
+    # Executing integration tests
+    GODEBUG=asyncpreemptoff=1 go test $test_path_non_parallel_group_1 -p 1 --integrationTest -v --testbucket=$BUCKET_NAME_NON_PARALLEL_GROUP_1 --testInstalledPackage=$run_e2e_tests_on_package -timeout $INTEGRATION_TEST_TIMEOUT
+    exit_code_non_parallel_group_1=$?
+    if [ $exit_code_non_parallel_group_1 != 0 ]; then
+      test_fail_np_group_1=$exit_code_non_parallel_group_1
+      echo "test fail in non parallel in group 1: " $test_fail_np_group_1
+    fi
+  done
+  return $test_fail_np_group_1
+}
+
+# Non parallel execution of integration tests located within specified test directories for group 2.
+function run_non_parallel_tests_group_2() {
```

Comment:
Created separate function as calling same function with argument in background was overriding the value.

FYI @raj-prince @sethiay

Reply:
@Tulsishah is $test_fail_np_group_1 value getting overridden if we use same function ? if that's the case then can we append a prefix/suffix of bucket name in this variable name and keep the function same? will it help ?

Reply:
`test_dir_np_group_1` the array itself is overriding, so all the tests are not executing that's what I face. It's not about failure code, it's about execution itself.

Reply:
It is fixed. Thanks @ankitaluthra1 I was capturing arrays properly in functions that's why it was happening.